src/mlpack/core/data/load_csv.hpp

```diff
@@ -263,28 +396,49 @@ class LoadCSV
     inFile.clear();
     inFile.seekg(0, std::ios::beg);
 
-    auto setCharClass = [&](iter_type const &iter)
+    while (std::getline(inFile, line))
     {
-      std::string str(iter.begin(), iter.end());
-      if (str == "\t")
+      // Remove whitespaces from either side
+      trim(line);
+
+      std::stringstream line_stream;
+      std::string token;
+
+      if(line.size() == 0)
       {
-        str.clear();
+        break;
       }
-      boost::trim(str);
 
-      inout(row, col++) = infoSet.template MapString<T>(std::move(str), row);
-    };
+      line_stream.clear();
+      line_stream.str(line);
 
-    while (std::getline(inFile, line))
-    {
-      // Remove whitespace from either side.
-      boost::trim(line);
+      while(line_stream.good())
+      {
+        if(token == "\t")
```

Comment:
`"\t"` doesn't refer to a tab character; but `'\t'` does.  I'm not sure I understand what this line is supposed to do?